runtime/interpreter/value.go

```diff
@@ -17936,9 +17943,7 @@ func (v *CompositeValue) ForEachFieldName(
 	f func(fieldName string) (resume bool),
 ) {
 	iterate := func(fn atree.MapElementIterationFunc) error {
-		return v.dictionary.IterateKeys(
-			StringAtreeValueComparator,
-			StringAtreeValueHashInput,
+		return v.dictionary.IterateReadOnlyKeys(
```

Comment:
Not sure if it's guaranteed that all uses of `CompositeValue.ForEachFieldName` are only read-only.

Reply:
@turbolent Good catch! I reverted to use non-readonly iterator here (and below for other comment).